test/types/reply.test-d.ts

```diff
@@ -29,10 +29,28 @@ const getHandler: RouteHandlerMethod = function (_request, reply) {
   expectType<(fullfilled: () => void, rejected: (err: Error) => void) => void>(reply.then)
 }
 
-const typedHandler: RouteHandler<{ Reply: { test: boolean } }> = async (request, reply) => {
-  expectType<(<T = { test: boolean; }>(payload?: T) => FastifyReply<RawServerDefault, RawRequestDefaultExpression<RawServerDefault>, RawReplyDefaultExpression<RawServerDefault>, { Reply: { test: boolean } }>)>(reply.send)
+interface ReplyPayload {
+  Reply: {
+    test: boolean
+  }
+}
+
+const typedHandler: RouteHandler<ReplyPayload> = async (request, reply) => {
+  expectType<(<T = ReplyPayload['Reply']>(payload?: T) => FastifyReply<RawServerDefault, RawRequestDefaultExpression<RawServerDefault>, RawReplyDefaultExpression<RawServerDefault>, ReplyPayload>)>(reply.send)
```

Comment:
What does the ` ReplyPayload['Reply']` syntax mean?

Reply:
If you have an interface and you need to access to some of the fields defined inside it, you have to use the `['']` syntax instead of the dot `.`. Here we are saying that if no T is passed it will default to `ReplyPayload['Reply'] type`. Usually it works but it doesn't work here.